tests/test_sendinblue_backend.py

```diff
@@ -0,0 +1,444 @@
+        error_response = b"""{
+            "code": "invalid_parameter",
+            "message": "valid sender email required"
+        }"""
+        self.set_mock_response(status_code=400, raw=error_response)
+        with self.assertRaises(AnymailAPIError) as cm:
+            self.message.send()
+        err = cm.exception
+        self.assertIn("code", str(err))
+        self.assertIn("message", str(err))
+
+        # No content in the error response:
+        self.set_mock_response(status_code=502, raw=None)
+        with self.assertRaises(AnymailAPIError):
+            self.message.send()
+
+
+class SendinBlueBackendAnymailFeatureTests(SendinBlueBackendMockAPITestCase):
+    """Test backend support for Anymail added features"""
+
+    def test_metadata(self):
+        self.message.metadata = {'user_id': "12345", 'items': 6, 'float': 98.6, 'long': longtype(123)}
+
+        with self.assertRaises(AnymailUnsupportedFeature):
+            self.message.send()
+
+    def test_send_at(self):
+        utc_plus_6 = get_fixed_timezone(6 * 60)
+        utc_minus_8 = get_fixed_timezone(-8 * 60)
+
+        with override_current_timezone(utc_plus_6):
+            # Timezone-aware datetime converted to UTC:
+            self.message.send_at = datetime(2016, 3, 4, 5, 6, 7, tzinfo=utc_minus_8)
+
+            with self.assertRaises(AnymailUnsupportedFeature):
+                self.message.send()
+
+    def test_tag(self):
+        self.message.tags = ["receipt"]
+        self.message.send()
+        data = self.get_api_call_json()
+        self.assertCountEqual(data['headers']["X-Mailin-tag"], "receipt")
+
+    def test_multiple_tags(self):
+        self.message.tags = ["receipt", "repeat-user"]
+        with self.assertRaises(AnymailUnsupportedFeature):
+            self.message.send()
+
+    def test_tracking(self):
+        # Test one way...
+        self.message.track_clicks = False
+        self.message.track_opens = True
+
+        with self.assertRaises(AnymailUnsupportedFeature):
+            self.message.send()
+
+        # ...and the opposite way
+        self.message.track_clicks = True
+        self.message.track_opens = False
+        with self.assertRaises(AnymailUnsupportedFeature):
+            self.message.send()
+
+    def test_template_id(self):
+        # SendinBlue use incremental ID to identify templates
+        self.message.template_id = "12"
+        self.message.send()
+
+        # We can't verify if the Url has changed
+
+    def test_template_id_with_empty_body(self):
+        message = mail.EmailMessage(from_email='from@example.com', to=['to@example.com'])
+        message.template_id = "9"
+        message.send()
+        data = self.get_api_call_json()
+        self.assertNotIn('htmlcontent', data)
+        self.assertEquals(data['textContent'], '')  # neither text nor html body
+        self.assertNotIn('subject', data)
+
+    def test_merge_data(self):
+        self.message.from_email = 'from@example.com'
```

Comment:
For most backends, this is a combined test for `message.merge_data` and `message.merge_global_data`. Since SendinBlue only supports the global variant, it'd be good to split that out (or roll it into the template test earlier), and verify that merge_global_data ends up as SiB "attributes". [Example from Postmark tests](https://github.com/anymail/django-anymail/blob/v1.4/tests/test_postmark_backend.py#L362-L377)

Reply:
Done